tests/cli.test.js

```diff
@@ -0,0 +1,607 @@
+
+        expectStringContains(stdout, 'Converting 6 images (lossy)...');
+        expectRatio(stdoutRatio, 55, 60);
+        expectFileNotModified(`${fileBasename}.png`);
+        expectFileExists(`${fileBasename}.webp`);
+      });
+    });
+
+    describe('Lossless (--lossless)', () => {
+      test('JPEG should be converted', () => {
+        const file = 'jpeg-one-pixel.jpg';
+        const stdout = runCliWithParams(`--webp --lossless ${workDir}${file}`);
+
+        expectFileRatio({ stdout, file, maxRatio: 80, minRatio: 75, outputExt: 'webp' });
+        expectFileNotModified(file);
+      });
+
+      test('PNG should be converted', () => {
+        const file = 'png-not-optimized.png';
+        const stdout = runCliWithParams(`--webp --lossless ${workDir}${file}`);
+
+        expectFileRatio({ stdout, file, maxRatio: 45, minRatio: 40, outputExt: 'webp' });
+        expectFileNotModified(file);
+      });
+
+      test('GIF should be converted', () => {
+        const file = 'gif-not-optimized.gif';
+        const stdout = runCliWithParams(`--webp --lossless ${workDir}${file}`);
+
+        expectFileRatio({ stdout, file, maxRatio: 25, minRatio: 20, outputExt: 'webp' });
+        expectFileNotModified(file);
+      });
+
+      test('Files should not be converted if ratio <= 0', () => {
+        const fileBasename = 'jpeg-low-quality';
+        const stdout = runCliWithParams(`--webp --lossless ${workDir}${fileBasename}.jpg`);
+
+        expectStringContains(stdout, 'Converting 1 image (lossless)...');
+        expectStringContains(stdout, 'Done!');
+        expectFileNotModified(`${fileBasename}.jpg`);
+        expectFileNotExists(`${fileBasename}.webp`);
+      });
+
+      test('Files in provided directory should be converted', () => {
+        const fileBasename = 'png-not-optimized';
+        const stdout = runCliWithParams(`--webp --lossless ${workDir}`);
+        const stdoutRatio = grepTotalRatio(stdout);
+
+        expectStringContains(stdout, 'Converting 6 images (lossless)...');
+        expectRatio(stdoutRatio, 1, 5);
+        expectFileNotModified(`${fileBasename}.png`);
+        expectFileExists(`${fileBasename}.webp`);
+      });
+    });
+  });
+
+  describe('Converting to AVIF and WebP at the same time (--avif --webp)', () => {
+    describe('Lossy', () => {
+      test('AVIF and WebP should be created', () => {
+        const fileBasename = 'png-not-optimized';
+        const stdout = runCliWithParams(`--avif --webp ${workDir}${fileBasename}.png`);
+        const stdoutRatio = grepTotalRatio(stdout);
+
+        expectStringContains(stdout, 'Converting 1 image (lossy)...');
+        expectStringContains(stdout, path.join(tempDir, `${fileBasename}.png`));
+        expectRatio(stdoutRatio, 85, 90);
+        expectFileNotModified(`${fileBasename}.png`);
+        expectFileExists(`${fileBasename}.avif`);
+        expectFileExists(`${fileBasename}.webp`);
+      });
+
+      // TODO: Fix a bug for this test case
```

Comment:
@343dev, will it be done in a separated PR, or here?

Reply:
Fixed it here.